cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/EquivalentInjectionConversion.java

```diff
@@ -21,55 +19,128 @@
 
     private static final String REGULATION_TARGET = "regulationTarget";
 
-    public EquivalentInjectionConversion(PropertyBag sm, Context context) {
-        super("EquivalentInjection", sm, context);
+    public EquivalentInjectionConversion(PropertyBag ei, Context context) {
+        super("EquivalentInjection", ei, context);
+    }
+
+    @Override
+    public void convertInsideBoundary() {
+        if (context.config().convertBoundary()) {
+            if (valid()) {
+                convert();
+            }
+        } else {
+            // If we find an Equivalent Injection at a boundary
+            // we will decide later what to do with it
+            //
+            // If finally a dangling line is created at the boundary node
+            // and the equivalent injection is regulating voltage
+            // we will have to transfer regulating voltage data
+            // from the equivalent injection to the dangling line
+            context.boundary().addEquivalentInjectionAtNode(this.p, nodeId());
+        }
     }
 
     @Override
     public void convert() {
+        // An equivalent injection found inside a modeling authority data
+        // will be mapped to a Generator
+        convertToGenerator();
+    }
+
+    // A dangling line has been created at the boundary node of the equivalent injection
+    public DanglingLine convertOverDanglingLine(DanglingLineAdder adder, PowerFlow fother) {
+        Regulation regulation = getRegulation();
+        DanglingLine dl;
+        if (regulation.status) {
+            // If this equivalent injection is regulating voltage,
+            // map it over the dangling line 'virtual generator'
+            dl = adder
+                    .setP0(fother.p())
+                    .setQ0(fother.q())
+                    .add();
+            dl.newGeneration()
```

Comment:
It's a bit weird: why the generation is not created using the adder, but using the DanglingLine?

Reply:
I used this design (similar to reactive limits for example) because the dangling line object can be consistent without it (contrary to shunt model or T3W's legs for example). Do you think it would be better to put it in DLAdder?